homeassistant/components/hausbus/config_flow.py

```diff
@@ -0,0 +1,113 @@
+from pyhausbus.HomeServer import HomeServer
+from pyhausbus.IBusDataListener import IBusDataListener
+from pyhausbus.ObjectId import ObjectId
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_SCHEMA = vol.Schema({})
+
+
+class ConfigFlow(IBusDataListener, config_entries.ConfigFlow, domain=DOMAIN):  # type: ignore[misc]
+    """Handle a config flow for hausbus."""
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._found_device = False
+        self._search_task: asyncio.Task | None = None
+        self.home_server = HomeServer()
+        self.home_server.addBusEventListener(self)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            # start searching for devices
+            return await self.async_step_wait_for_device()
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_SCHEMA, errors=errors
+        )
+
+    async def async_step_wait_for_device(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Wait for a hausbus device to be found."""
+        if not self._search_task:
+            self._search_task = self.hass.async_create_task(
+                self._async_wait_for_device()
+            )
+
+        if not self._search_task.done():
+            return self.async_show_progress(
+                step_id="wait_for_device",
+                progress_action="wait_for_device",
+                progress_task=self._search_task,
+            )
+
+        try:
+            await self._search_task
+        except TimeoutError:
+            return self.async_show_progress_done(next_step_id="search_timeout")
+        finally:
+            self._search_task = None
+
+        return self.async_show_progress_done(next_step_id="search_complete")
+
+    async def async_step_search_timeout(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Inform the user that no device has been found."""
+        if user_input is not None:
+            return await self.async_step_wait_for_device()
+
+        self._set_confirm_only()
+        return self.async_show_form(step_id="search_timeout")
+
+    async def async_step_search_complete(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Create a configuration entry for the hausbus devices."""
+        return self.async_create_entry(title="Haus-Bus", data={})
+
+    async def _async_wait_for_device(self) -> None:
+        """Start searching for devices and wait until at least one device was found or timeout is reached."""
+        self.home_server.searchDevices()
```

Comment:
I guess this sends UDP packages, so should be run in an executor thread, not an event loop?

Reply:
The event loop is handled in the pyhausbus library.

Reply:
Can you tell me how to access the hass executor thread from here? In __init__.py I had access to the hass object to add the executor job.

Reply:
The hass object is available as `self.hass`